File: Bit Manipulation/SubSet.java
// print all the possible subsets of the array

public static void subSets(String[] arr, int n) {
    for (int i = 0; i < Math.pow(2, n); i++) {
        for (int j = 0; j < n; j++) {
            if ((i & (1 << j)) > 0) {
                System.out.print(arr[j] + " ");
            }
        }
        System.out.println();
    }
}

File: Bit Manipulation/SubSet.js
// print all the possible subsets of the array

function subSets(arr) {
    for (let i = 0; i < Math.pow(2, arr.length); i++) {
        for (let j = 0; j < arr.length; j++) {
            if ((i & (1 << j)) > 0) {
                process.stdout.write(arr[j] + " ");
            }
        }
        console.log();
    }
}
